nmagent/responses.go

```diff
@@ -38,7 +38,19 @@ type NCVersionList struct {
 }
 
 type AzResponse struct {
-	HomeAz uint `json:"homeAz"`
+	HomeAz     uint `json:"homeAz"`
+	APIVersion uint `json:"apiVersion"`
+}
+
+func (az AzResponse) Valid() bool {
+	// 0 should be valid when NMA version is old and does not have the apiVersion value in home az response
+	//nolint:gomnd // these magic numbers are made by nma design
+	return az.APIVersion == 0 || az.APIVersion == 2
+}
+
+func (az AzResponse) NmaAppliedTheIPV6Fix() bool {
+	//nolint:gomnd // this magic number is made by nma design
+	return az.APIVersion == 2
```

Comment:
Shouldn't this be az.APIVersion >= 2 instead of az.APIVersion == 2 since it can change in future?

Reply:
Vishewesh mentioned that for now we only want to allow APIVersion as 2, and he want us to raise alerts if we get APIVersion other than 2.  from what he means, when we get APIVersion as "3", we should mark it as an invalid version number and then raise alert on it. That way, we may not want it to indicate the fix is applied.

Reply:
Since this API version field is being introduced as general purpose field which can be used for future bugs also, we should account for what would be AZ behavior if it becomes 3 some day.